extkingpin/pathorcontent.go

```diff
@@ -38,24 +39,25 @@ type FlagClause interface {
 // RegisterPathOrContent registers PathOrContent flag in kingpinCmdClause.
 func RegisterPathOrContent(cmd FlagClause, flagName string, help string, opts ...Option) *PathOrContent {
 	fileFlagName := fmt.Sprintf("%s-file", flagName)
-	contentFlagName := flagName
-
-	fileHelp := fmt.Sprintf("Path to %s", help)
-	fileFlag := cmd.Flag(fileFlagName, fileHelp).PlaceHolder("<file-path>").String()
-
-	contentHelp := fmt.Sprintf("Alternative to '%s' flag (mutually exclusive). Content of %s", fileFlagName, help)
-	contentFlag := cmd.Flag(contentFlagName, contentHelp).PlaceHolder("<content>").String()
+	fileFlag := cmd.Flag(fileFlagName, fmt.Sprintf("Path to %s", help)).PlaceHolder("<file-path>")
+	contentFlag := cmd.Flag(flagName, fmt.Sprintf("Alternative to '%s' flag (mutually exclusive). Content of %s", fileFlagName, help)).PlaceHolder("<content>")
 
 	p := &PathOrContent{
-		flagName:        flagName,
-		path:            fileFlag,
-		content:         contentFlag,
-		required:        false,
-		envSubstitution: false,
+		flagName: flagName,
 	}
+
 	for _, opt := range opts {
 		opt(p)
 	}
+
+	if p.hidden {
+		p.path = fileFlag.Hidden().String()
+		p.content = contentFlag.Hidden().String()
```

Comment:
If you don't call `.String()` here, you won't need the `return p` on line 56, because the `.String()` call comes just below.

Reply:
`p.path/content` is of type `*string`, and the `file/contentFlag` variables are of type `*kingpin.FlagClause`. The `Hidden()` call also returns the same type, so we need to call `String()` which returns `*string` here and return the struct early. 🙂

Reply:
You can mutate `fileFlag` and `contentFlag`:

```go
if p.hidden {
    fileFlag = fileFlag.Hidden()
    contentFlag = contentFlag.Hideen()
}
p.path = fileFlag.String()
p.content = contentFlag.String()
return p
```